yt/data_objects/static_output.py

```diff
@@ -306,17 +311,12 @@ def backup_filename(self):
         name, _ext = os.path.splitext(self.filename)
         return name + "_backup.gdf"
 
-    @property
-    def unique_identifier(self):
-        if self._unique_identifier is None:
-            self._unique_identifier = int(os.stat(self.parameter_filename)[ST_CTIME])
-            name_as_bytes = bytearray(map(ord, self.parameter_filename))
-            self._unique_identifier += fnv_hash(name_as_bytes)
-        return self._unique_identifier
-
-    @unique_identifier.setter
-    def unique_identifier(self, value):
-        self._unique_identifier = value
+    @cached_property
+    def unique_identifier(self) -> str:
+        retv = int(os.stat(self.parameter_filename)[ST_CTIME])
+        name_as_bytes = bytearray(map(ord, self.parameter_filename))
+        retv += fnv_hash(name_as_bytes)
+        return str(retv)
```

Comment:
An important note here is that I'm changing the `unique_identifier` from a mutable property to an immutable one (cached properties can't be mutable). The previous implementation was from https://github.com/yt-project/yt/pull/2559, where my main concern was to reduce duplication, but leaving this property as a mutable was just an implementation detail, not a thoughtful design choice. Now I think it _should_ be immutable.

Reply:
I agree with that.